third-party/bsp/stmf7cube/arch.c

```diff
@@ -69,11 +74,58 @@ static void SystemClock_Config(void)
   }
 }
 
+#if ENABLE_CPU_CACHE
+static void cpu_cache_enable(void) {
+	SCB_EnableICache();
+	SCB_EnableDCache();
+}
+
+static void mpu_config(void) {
+	MPU_Region_InitTypeDef MPU_InitStruct;
+
+	/* Disable the MPU */
+	HAL_MPU_Disable();
+
+	/* Write Through mode */
+	MPU_InitStruct.Enable = MPU_REGION_ENABLE;
+	MPU_InitStruct.AccessPermission = MPU_REGION_FULL_ACCESS;
+	MPU_InitStruct.IsBufferable = MPU_ACCESS_NOT_BUFFERABLE;
+	MPU_InitStruct.IsCacheable = MPU_ACCESS_CACHEABLE;
+	MPU_InitStruct.IsShareable = MPU_ACCESS_NOT_SHAREABLE;
+	MPU_InitStruct.TypeExtField = MPU_TEX_LEVEL0;
+	MPU_InitStruct.SubRegionDisable = 0x00;
+	MPU_InitStruct.DisableExec = MPU_INSTRUCTION_ACCESS_ENABLE;
+
+	/* Configure the MPU attributes as WT for SRAM
```

Comment:
It might be worth noting that certain Cortex-M7 revisions have [errata](https://static.docs.arm.com/epm064408/80/Cortex-M7_Software_Developers_Errata_Notice_v8.pdf#page=11) related to the use of write-through cache.

Reply:
Thanks!

We are having problems with stm32f7629i discovery now, maybe due caches, but I am still not sure. That's why this PR is WIP.

Generally, we would like to avoid using WT cache mode, because it is slower than write-back. So it is temporary solution derived from STM32Cube.

Reply:
After your comment, I am almost sure we should avoid WT cache :)